src/com/google/javascript/jscomp/PeepholeReplaceKnownMethods.java

```diff
@@ -59,9 +60,11 @@ Node optimizeSubtree(Node subtree) {
   private Node tryFoldKnownMethods(Node subtree) {
     // For now we only support string methods .join(),
     // .indexOf(), .substring() and .substr()
+    // array method concat()
     // and numeric methods parseInt() and parseFloat().
 
     subtree = tryFoldArrayJoin(subtree);
+    subtree = tryToFoldArrayConcat(subtree);
```

Comment:
Not sure if it the right place to put this function call,
because below here lots of nested if statement, and not clear if we should perform only one optimization during `tryFoldKnownMethods` subtree call or several optimizations or probably `tryFoldKnownMethods` may be refactored.